service_manual_service_standard migration

AshGDS opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What / Why

- Migrates `/service-manual/service-standard` from `government-frontend` to `frontend`
- Heroku deployment link: https://govuk-frontend-app-pr-4796.herokuapp.com/service-manual/service-standard
- Live site link: https://www.gov.uk/service-manual/service-standard
- [Trello card](https://trello.com/c/rSmP2fUD/597-move-servicemanualservicestandard-from-government-frontend-to-frontend), [Jira issue PNP-6371](https://gov-uk.atlassian.net/browse/PNP-6371)
- HTML Diff check: https://www.diffchecker.com/K7LWD0Ba/

## Visual changes
- The new page will have the full size footer.
- The new page will no longer have the phase banner (this will be removed when the content item is updated in the publishing app.)

## Relevant PRs
- Publishing app: https://github.com/alphagov/service-manual-publisher/pull/1765
- Removal in `government-frontend`: https://github.com/alphagov/government-frontend/pull/3683

AshGDS committed:
Add service_manual_service_standard to govuk_examples.yml & README.md

KludgeKML commented on app/views/service_manual_service_standard/index.html.erb:
```diff
@@ -0,0 +1,59 @@
+<% add_view_stylesheet("service-manual-service-standard") %>
+<% content_for :extra_headers do %>
+  <meta name="description" content="<%= strip_tags(@content_item.description) %>">
```
This can be content_item (without the @)

KludgeKML commented on app/views/service_manual_service_standard/index.html.erb:
```diff
@@ -0,0 +1,59 @@
+  <meta name="description" content="<%= strip_tags(@content_item.description) %>">
+  <%= render "govuk_publishing_components/components/machine_readable_metadata", { content_item: content_item.to_h, schema: :article } %>
+<% end %>
+<%= content_for :title, "#{@content_item.title} - Service Manual - GOV.UK" %>
```
This can be content_item (without the @)

KludgeKML commented on app/presenters/service_manual_service_standard_presenter.rb:
```diff
@@ -0,0 +1,63 @@
+    @content_item = content_item
+  end
+
+  def points
```
I think the points attribute might be better in the model rather than a presenter. (we could also memoize the loaded and sorted list)

KludgeKML commented on app/views/shared/_email_signup.html.erb:
```diff
@@ -0,0 +1,21 @@
+<div class="related-item govuk-!-padding-top-4">
+  <%= render "govuk_publishing_components/components/heading", {
```
Depending on how many other views we expect to use it. 

If we think more views will use it, I'd suggest passing in the link as a local, rather than relying on the presence/ duck type of `@presenter`

AshGDS replied:
Thanks - yeah, I'll decouple it from `@presenter`. This partial is also used by `service_manual_topic` (which hasn't been migrated yet) so should we keep it in `shared`?

KludgeKML replied:
(I see it's been moved into the views/service_manual folder, so this is resolved)

KludgeKML commented on spec/system/service_manual_service_standard_spec.rb:
```diff
@@ -0,0 +1,64 @@
+      visit base_path
+    end
+
+    it "service standard page has a heading, summary and intro" do
```
asserts in this spec need to be updated to expectations.

KludgeKML reviewed (changes requested):
A few notes about structure, and we should probably chat with @andysellick about whether this should have its own controller or be in the existing service manual controller.

AshGDS commented on app/views/shared/_service_manual_breadcrumbs.html.erb:
```diff
@@ -0,0 +1,8 @@
+<div class="govuk-grid-row">
```
@KludgeKML I put it in `shared` as when I looked in `government-frontend` it was used in `service_manual_topic` and `service_manual_guide` which haven't been migrated yet - was I planning ahead to early by putting it in `shared`?

KludgeKML replied:
If we're moving this into the same controller that service manual topics and guides will be in, then it can be a partial that's in the app/views/service_manual folder (ie in the folder with the handful of views from one controller that use it rather than this folder, which is for partials shared by multiple controllers)

AshGDS commented:
Thanks @KludgeKML - I've think I've responded to every comment and fixed everything, apart from the question about the controller and the shared partials.

KludgeKML commented on spec/system/service_manual_service_standard_spec.rb:
```diff
@@ -0,0 +1,64 @@
+RSpec.describe "Field of operation page" do
```
Oh, missed that this name hadn't been updated.

AshGDS committed:
Add placeholder controller/view/route for service_manual_service_standard

AshGDS committed:
Copy service_manual_service_standard view from government-frontend

Audit trail: https://github.com/alphagov/government-frontend/blob/adbbba65aa8355dfed0970c12749f5c34e16c81b/app/views/content_items/service_manual_service_standard.html.erb

AshGDS committed:
Copy service_manual_breadcrumbs partial from government-frontend

- We also need to turn off contextual breadcrumbs for this document type
  in the application helper, and mark that the breadcrumbs partial is
  for the before_content block in the layout (there's no header block
  in frontend).

Audit trail: https://github.com/alphagov/government-frontend/blob/62fb6800ec434a4a1176c7513038e61cb9da2dc2/app/views/content_items/_service_manual_breadcrumbs.html.erb

AshGDS committed:
Copy email_signup partial from government-frontend

Audit trail: https://github.com/alphagov/government-frontend/blob/03169f2d93f105aed12c458e256427fb23317453/app/views/shared/_email_signup.html.erb

AshGDS committed:
Copy service_manual_service_standard presenter from government-frontend

Removes the 'poster' functionality, as that's no longer used.

Audit trail: https://github.com/alphagov/government-frontend/blob/94416eb1636589415edd208ec67ea3494a2e3e24/app/presenters/service_manual_service_standard_presenter.rb

AshGDS committed:
Modify view and presenter to make view render

- Adds body function in presenter
- Fixes content_item references in presenter
- Removes phase banner in view
- Fixes content_item/presenter references in view
- Removes unused poster HTML in view
- Adds description metatags and machine readable data
- Adds GOV.UK to the page title
- Scope slimmer_template change to index only

AshGDS committed:
Migrate service_manual_service_standard stylesheets from government-frontend

Audit trails:

https://github.com/alphagov/government-frontend/blob/e24ea2f63e90120078325f376d35aa623f26895f/app/assets/stylesheets/modules/_page-header.scss

https://github.com/alphagov/government-frontend/blob/e24ea2f63e90120078325f376d35aa623f26895f/app/assets/stylesheets/modules/_service-standard-point.scss

AshGDS committed:
Move points generator to model

AshGDS committed:
Add service_manual_service_standard request tests

AshGDS committed:
Add service_manual_service_standard system tests

Audit trail: https://github.com/alphagov/government-frontend/blob/adbbba65aa8355dfed0970c12749f5c34e16c81b/test/integration/service_manual_service_standard_test.rb

AshGDS commented:
@KludgeKML I've moved the controller and those partials into `service_manual` now, thanks

KludgeKML reviewed (approved):
LGTM!
